packages/components/src/presentation-components/properties/inputs/QuantityPropertyEditorInput.tsx

```diff
@@ -72,7 +72,24 @@ const QuantityPropertyValueInput = forwardRef<PropertyEditorAttributes, Quantity
         });
     };
 
-    return <Input size="small" {...inputProps} ref={inputRef} onBlur={onBlur} />;
+    useEffect(() => {
+      if (setFocus) {
+        setTimeout(() => {
+          inputRef.current && inputRef.current.focus();
+        }, 0);
+      }
+    }, [setFocus]);
+
+    return (
+      <Input
+        size="small"
+        {...inputProps}
+        disabled={propertyRecord.isReadonly || inputProps.disabled}
+        ref={inputRef}
+        onBlur={onBlur}
+        onFocus={() => inputRef.current?.setSelectionRange(0, 9999)}
```

Comment:
Why not `inputRef.current?.setSelectionRange(0, inputProps.value.length)`?

Reply:
Changed

Reply:
Reverted changed for `QuantityPropertyEditor`, the value is provided from a custom hook meaning the value is 0 (length) on first load when we focus on the component, left the adjusted selection for the numerical value. I think it is fine to use constant value for selection range as trying to use raw value + const margin might couse bugs.

Reply:
changed